### PR TITLE
ENH: include conversion to nullable float in convert_dtypes()

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/38110

There is one potentially corner case: what with floats that are all "integer"-like? I think we want to keep returning nullable int for that, at least by default, and that is what I did now in this PR But we might want to add a parameter controlling that behaviour? (but that can also be added later on if there is demand for it)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Also, note, this builds upon https://github.com/pandas-dev/pandas/pull/38113. Once that is merged, the changes I needed to do to the test cases will become clear.

File: pandas/core/dtypes/cast.py
Comment (jreback) [inline pandas/core/dtypes/cast.py:1276]: needs a version added, 
liking less and less the fact that we have many options here.

would change to an include=, exclude interface. (might involve deprecating the existing options). am -1 on doing this for 1.2 for this reason.

**Review by jreback — CHANGES_REQUESTED** [review]
this is after the feature freeze, so moving off 1.2

File: pandas/core/dtypes/cast.py
Comment (jorisvandenbossche) [reply pandas/core/dtypes/cast.py:1276]: > needs a version added,

This is the internal method, so doesn't need a versionadded. The public method already has a versionadded

**Comment by jorisvandenbossche** [comment]
> liking less and less the fact that we have many options here.
>
>would change to an include=, exclude interface. (might involve deprecating the existing options). am -1 on doing this for 1.2 for this reason.

We had a long discussion about this in the original PR (https://github.com/pandas-dev/pandas/pull/30929), and I want to remind that it was actually you who asked for the separate options .. (I personally think we should not have added ay of those options in the first place)

If we want to change the interface of `convert_dtypes`, we can do that later as well. I don't think that needs to block this PR. I could also leave out the option for now (and only have the default behaviour of True)

**Comment by jreback** [comment]
and am happy to revisit but not at the 11hour

**Comment by jorisvandenbossche** [comment]
I am happy with leaving out the option, but I think we should still include the actual functionality in 1.2. The float EA array already is included in 1.2, this just enables it in the `convert_dtypes` (as is documented that this function will change when new nullable dtypes get added).

**Review by jreback — CHANGES_REQUESTED** [review]
ok this is fine (we should change the api but that's a future issue, do we have one?)

is there a test for convert_floating=True & convert_integer=True and it converts floats to ints?

otherwise lgtm

File: pandas/core/generic.py
Comment (jreback) [inline pandas/core/generic.py:6185]: might be worth mentioning here that this changed in 1.2

**Comment by jorisvandenbossche** [comment]
> we should change the api but that's a future issue, do we have one?

No, but I will open one.

> is there a test for convert_floating=True & convert_integer=True and it converts floats to ints?

Yes, all possible combo's of all parameters are tested, and we have cases of both integer-like floats as well as actual floats that can't be casted to int

File: pandas/tests/series/methods/test_convert_dtypes.py
Comment (jorisvandenbossche) [inline pandas/tests/series/methods/test_convert_dtypes.py:63]: @jreback this is the test case with floats that are int-like. By default it gets casted to Int64, but if `convert_integer=False` to Float64

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Pull request merged**
